libyul/optimiser/LoadResolver.cpp

```diff
@@ -65,15 +65,12 @@ void LoadResolver::tryResolve(
 		return;
 
 	YulString key = std::get<Identifier>(_arguments.at(0)).name;
-	if (
-		_location == StoreLoadLocation::Storage &&
-		m_storage.values.count(key)
-	)
-		_e = Identifier{locationOf(_e), m_storage.values[key]};
-	else if (
-		m_optimizeMLoad &&
-		_location == StoreLoadLocation::Memory &&
-		m_memory.values.count(key)
-	)
-		_e = Identifier{locationOf(_e), m_memory.values[key]};
+	if (_location == StoreLoadLocation::Storage)
+	{
+		if (auto value = util::fetchValue(m_storage, key))
+			_e = Identifier{locationOf(_e), *value};
+	}
+	else if (m_optimizeMLoad && _location == StoreLoadLocation::Memory)
```

Comment:
This is also somehow different. Before, if the location is storage but the key isn't found, it would try the `else if`. Now in that case it won't.

Reply:
It would have entered the ``else if``, but since the location would be storage, ``_location == StoreLoadLocation::Memory`` would be false, i.e. the ``else if`` would not be entered, i.e. we might as well not even enter it.

Reply:
Yea ok